### PR TITLE
*: Upgrade golangci-lint

### PR DESCRIPTION
**Pull request by kakkoyun**
Signed-off-by: Kemal Akkoyun <kakkoyun@gmail.com>

### CODE REVIEW

File: .github/workflows/golangci-lint.yml [hunk 1]
```diff
@@ -6,10 +6,12 @@ on:
     branches: [ main ]
     paths:
       - '**.go'
+      - .golangci.yml
   pull_request:
     branches: [ main ]
     paths:
       - '**.go'
+      - .golangci.yml
```
[the remark this answers is not in the document]
Comment (kakkoyun) [reply .github/workflows/golangci-lint.yml:14]: nvm

**Review by metalmatze — APPROVED** [review]
Any particular reason to upgrade?

**Comment by kakkoyun** [comment]
Go1.18 doesn't work with the older versions. I happen to upgrade things on my side.

**Comment by kakkoyun** [comment]
I think github actions are broken right now :thinking:

**Pull request merged**